harness/determined/core/_checkpoint.py

```diff
@@ -302,7 +346,17 @@ def _upload_sharded(
 
         if want_upload:
             assert ckpt_dir
-            self._storage_manager.upload(src=ckpt_dir, dst=storage_id)
+            # If more than one uploading rank has a file of the same name and content,
+            # use the smallest rank to upload the file.
```

Comment:
I want to double-check: this is valid because if there were a true conflict, we wouldn't have gotten this far, right?

As in, this isn't actually a conflict resolution strategy, is it?  Is it just for deduplicating identical uploads?

Reply:
If there is a true conflict (i.e., md5 hashes are different for two files with the same names) then we will never get here. Exception will be thrown from `_try_resolving_conflicts`.

If we get here, we may still have multiple ranks having the same version of the file and we don't want to upload the same file multiple time. So yes, this is for `deduplicating identical uploads`.

Reply:
That explanation might be nice to include in the comment.  Otherwise it is a bit misleading to somebody who doesn't have global knowledge of the whole process.